test/uri.c

```diff
@@ -172,13 +172,15 @@ int test_uri_encode(void)
 
 int test_uri_user(void)
 {
+	re_fprintf(stderr, "Testing uri user");
 	const struct {
 		const char *enc;
 		const char *dec;
 	} uriv[] = {
 		{"alice%40atlanta.com", "alice@atlanta.com"},
 		{"project%20x", "project x"},
-		{"*21%23", "*21#"}
+		{"*21%23", "*21#"},
+		{"*21%23%C3%A4%E2%82%AC%40%20", "*21#ä€@ "}
```

Comment:
try to avoid Unicode characters in the source code.

you can use a binary string instead, i.e. "\x01\x02\x03\x04" etc.

Reply:
Thank you! So simple, and I was convinced it would solve the Windows problem. I am not sure what the MSVC compiler is doing here...

Do you have any more insight into what could be going wrong on the Windows machines? Or could you provide the script to setup Windows the way it is setup in the GitHub Actions?

Reply:
perhaps some C-string is not null terminated ?

add some debugging, see if the program is stuck in a while loop

Reply:
Yes looks like a out of bounds, this is the error window message, why windows test hangs:

```
Debug Assertion Failed!

Program: retest.exe File: minkernel\crts\ucrt\src\appcrt\convert\isctype.cpp Line: 36

Expression: c >= -1 && c <= 255
```
https://stackoverflow.com/questions/39538846/program-crashes

This is the Call stack:

```
retest.exe!is_unreserved(char c) Line 43	C
 	retest.exe!is_user(char c) Line 85	C
 	retest.exe!comp_escape(re_printf * pf, const pl * pl, bool(*)(char) eh) Line 173	C
 	retest.exe!uri_user_escape(re_printf * pf, const pl * pl) Line 235	C
 	retest.exe!re_vhprintf(const char * fmt, char * ap, int(*)(const char *, unsigned __int64, void *) vph, void * arg) Line 311	
 	retest.exe!mbuf_printf(mbuf * mb, const char * fmt, ...) Line 540	C
 	retest.exe!test_uri_user() Line 213	C
 	retest.exe!test_unit(const char * name, bool verbose) Line 506	C
 	retest.exe!test_reg(const char * name, bool verbose) Line 740	C
 	retest.exe!main(int argc, char * * argv) Line 235	C
``

Reply:
`char c` is `-61`

Reply:
Thank you! I'm looking into it right now.

Reply:
Oh I think `isalnum()` is the problem. It behaves differently on windows.

Maybe casting is enough, to avoid negative values:

```c
isalnum((unsigned char)c);
```

Reply:
I casted it because "ä" is represented by 0xC3 0xA4 in UTF-8 which as signed char gives us -61 and -92. So they are valid values, but need to be casted to be correctly interpreted by `isalnum`.